presentation/src/main/java/com/yapp/presentation/ui/member/qrcodescanner/QrCodeScanner.kt

```diff
@@ -0,0 +1,219 @@
+import android.util.Log
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.camera.core.CameraSelector
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST
+import androidx.camera.core.Preview
+import androidx.camera.lifecycle.ProcessCameraProvider
+import androidx.camera.view.PreviewView
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.core.content.ContextCompat
+import androidx.navigation.NavController
+import com.google.common.util.concurrent.ListenableFuture
+import com.yapp.common.theme.AttendanceTypography
+import com.yapp.presentation.R
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+
+@Composable
+fun QrCodeScanner(
+    modifier: Modifier = Modifier,
+    navController: NavController
+) {
+    val context = LocalContext.current
+    var hasCamPermission by remember {
+        mutableStateOf(
+            ContextCompat.checkSelfPermission(
+                context,
+                Manifest.permission.CAMERA
+            ) == PackageManager.PERMISSION_GRANTED
+        )
+    }
+    val launcher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.RequestPermission(),
+        onResult = { isGranted ->
+            hasCamPermission = isGranted
+        }
+    )
+    LaunchedEffect(key1 = true) {
+        launcher.launch(Manifest.permission.CAMERA)
+    }
+
+    Column(
+        modifier = Modifier
+            .fillMaxHeight()
+            .fillMaxWidth()
+    ) {
+        if (hasCamPermission) {
+            Surface {
+                CameraPreview()
+                ScannerDecoration(
+                    modifier = modifier,
+                    navController = navController
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun CameraPreview() {
+    val context = LocalContext.current
+    val lifecycleOwner = LocalLifecycleOwner.current
+    var preview by remember { mutableStateOf<Preview?>(null) }
+    var code = remember { mutableStateOf("") }
```

Comment:
```suggestion
    var code by remember { mutableStateOf("") }
```
위와 동일한 컨벤션으로 갖는게 좋을 것 같아요

Reply:
엇 보니 code 를 사용하는 곳이 없네요. 지우는게 어떨까요!

Reply:
도메인 로직 엮을 때 사용하게 될 것 같아서 만들어놓고 토스트만 띄우도록 사용하고 있었어요! 우선 제거해 두는 게 좋을까요?

Reply:
아항 ㅎㅎ 그럼 두는걸로!